mkinitcpio

```diff
@@ -251,6 +260,56 @@ build_image() {
     fi
 }
 
+build_uefi(){
+    set -x
+    local out=$1 microcode=$2 initramfs=$3 cmdline=$4 osrelease=$5 splash=$6 kernelimg=$7 uefistub=$8 errmsg=
+    OBJCOPYARGS=()
+
+    if [[ -z "$uefistub" ]]; then
+        for stub in {/usr,}/lib/systemd/boot/efi/linux{x64,ia32}.efi.stub {/usr,}/lib/gummiboot/linux{x64,ia32}.efi.stub; do
+            if [[ -f "$stub" ]]; then
+                uefistub="$stub"
+                break
+            fi
+        done
+    fi
+
+    if [[ -z "$kernelimg" ]]; then
+        for img in "/lib/modules/$KERNELIMAGE" "/boot/vmlinuz-$KERNELIMAGE" "/boot/vmlinuz-linuz"; do
+            if [[ -f "$img" ]]; then
+                KERNELIMAGE="$img"
+                break
+            fi
+        done
+    fi
+
+    if [[ -z "$cmdline" ]]; then
+        if [[ -f "/etc/kernel/cmdline" ]]; then
+            cmdline="/etc/kernel/cmdline"
+        else
+            cmdline="/proc/cmdline"
+        fi
+    fi
+
+    if [[ -n "$splash" ]]; then
+        OBJCOPYARGS+=("--add-section .splash=$splash --change-section-vma .splash=0x40000")
```

Comment:
This should probably be using the array like an array:

```
OBJCOPYARGS+=(--add-section .splash="$splash" --change-section-vma .splash=0x40000)
```

Reply:
Argh, I forget the syntax for these things.